### PR TITLE
Glm c log

### PR DESCRIPTION
**Pull request by gmcmacran**
- [ ] closes #xxxx
- [ ] tests added / passed. 
- [ ] code/documentation is well formatted.  
- [ ] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message). 

<details>

Please don't merge this pull request. I am making it to check style of code and execute tests. This branch will be rebased after tests and code style passes. 

This code adds the log complement link the binomial family for the generalized linear model.

### CODE REVIEW

**Comment by josef-pkt** [comment]
I never looked at this one

I found this article, it has a minus in front in the link function, so that linear predictor has to be positive
`g(p) = - log(1 - p)`

Piegorsch, Walter W. 1992. “Complementary Log Regression for Generalized Linear Models.” The American Statistician 46 (2): 94–99. https://doi.org/10.2307/2684172.

Do you have a specific reference that you used?

**Comment by josef-pkt** [comment]
skimming part of the article
CLog corresponds to right censored 0, 1, poisson with linear link
cloglog corresponds to censored poisson with log link

**Comment by gmcmacran** [comment]
> I never looked at this one
> 
> I found this article, it has a minus in front in the link function, so that linear predictor has to be positive `g(p) = - log(1 - p)`
> 
> Piegorsch, Walter W. 1992. “Complementary Log Regression for Generalized Linear Models.” The American Statistician 46 (2): 94–99. https://doi.org/10.2307/2684172.
> 
> Do you have a specific reference that you used?

It is out of Generalized Linear Models and Extensions 3rd edition by James W. Hardin and Joseph M. Hilbe.  Section 11.6.4 talks about the log complement. The formulas come out of tables in the appendix.  This book defines the link as ln(1 -u) = XB.

This stata documentation shows the link as I have it here.  See page 7 of https://www.stata.com/manuals/rglm.pdf.

**Comment by gmcmacran** [comment]
The cauchy link is now failing tests. I have not looked into it, but my guess is the slope of link function is very steep and the numerical approximation of the derivative is just a little off. 

@josef-pkt Considering I did not change that code, how did the tests pass on the master branch and not on this one? Let me know if you want me to fix this on this branch or keep this branched focused on the logc link.

File: docs/source/glm.rst [hunk 1]
```diff
@@ -180,6 +180,7 @@ available link functions can be obtained by
    CDFLink
    CLogLog
    LogLog
+   CLog
```
Comment (josef-pkt) [inline docs/source/glm.rst:183]: LogC

**Comment by josef-pkt** [comment]
cauchy now uses different random numbers because you inserted logc before cauchy in the loop
It might be better to set the seed inside the loop to make the random numbers independent of sequence

**Comment by josef-pkt** [comment]
LogC is a bit weird because it reverses the direction larger beta increases prob(0) and reduces mu.

We might want to add a `CLog` version in additionally to the `LogC` version with the usual monotonicity, negative in front as in the article I found.

I don't have the Hardin Hilbe GLM book and never looked at it.

File: docs/source/glm.rst (hunk 1 above)
Comment (gmcmacran) [reply docs/source/glm.rst]: Thanks for finding this. It is fixed now.

**Comment by gmcmacran** [comment]
Rebased in https://github.com/statsmodels/statsmodels/pull/8155.

Closing this pull request.

**Pull request closed without merging**